File: backend/src/main/java/org/example/backend/dto/WorkoutDayDto.java
package org.example.backend.dto;

import org.example.backend.model.Exercise;
import org.example.backend.model.MuscleGroup;
import org.example.backend.model.WorkoutDayType;

import java.time.DayOfWeek;
import java.util.List;
import java.util.Set;

public record WorkoutDayDto(DayOfWeek day,
                            WorkoutDayType type,
                            Set<MuscleGroup> targetMuscles,
                            List<Exercise> exercises) {
}

File: frontend/src/types/WorkoutDayDto.ts
import type {DayOfWeek} from "./DayOfWeek.ts";
import type {WorkoutDayType} from "./WorkoutDayType.ts";
import type {MuscleGroup} from "./MuscleGroup.ts";
import type {ExerciseDto} from "./ExerciseDto.ts";

export type WorkoutDayDto = {
    day: DayOfWeek,
    type: WorkoutDayType,
    targetMuscles: MuscleGroup[],
    exercises: ExerciseDto[];
}